### PR TITLE
feat: PBI-28 team authentication

### PR DESCRIPTION
**Pull request by andrew-susanto**
### Describe your changes
Implement team authentication as below:
- Add new permission and authentication classes
- Adjust current API query to team
- Add endpoint for current session team, available team, and change session team

### Backlog name
PBI-28: team authentication

### Acceptance criteria
- User able to login using user account and access to team data
- User able to change current session to another team

### Example Request
Get Current Team 
```
curl --location --request GET 'http://localhost:8000/auth/current_team/' \
--header 'Authorization: Token xxx' \
--data-raw ''
```

Get Available Team
```
curl --location --request GET 'http://localhost:8000/auth/available_team/' \
--header 'Authorization: Token xxx' \
--data-raw ''
```
```
[
    {
        "id": 3,
        "name": "Admin2"
    },
    {
        "id": 1,
        "name": "Admin"
    }
]
```

Change current session team
```
curl --location --request POST 'http://localhost:8000/auth/change_team/' \
--header 'Authorization: Token xxx' \
--header 'Content-Type: application/json' \
--data-raw '{
    "id": 3
}'
```
```
{
    "success": true
}
```
```
{
    "error": "Invalid team id"
}
```

### Checklist
- [x] I have performed a self-review of my code
- [x] Code successfully run on local
- [x] Feature / changes tested on local
- [x] No failing unit test
- [ ] Passed UAT Test
- [x] Sonarqube tested 
- [ ] Required any changes to environment variable

### CODE REVIEW

**Review by ferdifdi — APPROVED** [review]

**Review by luthfifahlevi — COMMENTED** [review]
Really brilliant!
And please check my comments ya, beberapa hal pengen ku-confirm wwkkw thank you

File: alerts/tests.py [hunk 1]
```diff
@@ -560,13 +581,16 @@ def test_when_api_monitor_failed_and_discord_enabled_then_send_alert(self, mock_
     @patch("django.contrib.auth.models.User.email_user")
     def test_when_api_monitor_failed_and_email_enabled_then_send_alert(self, mock_send_mail, mock_interrupt):
         user = User.objects.create_user(username='test', email='test@test.com', password='test123')
+        team = Team.objects.create(name='test team')
+        TeamMember.objects.create(team=team, user=user)
```
[the remark this answers is not in the document]
Comment (andrew-susanto) [reply alerts/tests.py:585]: wkwk kayaknya gakepake, otw delete, thank you 🙏

File: cron/tests.py [hunk 2]
```diff
@@ -141,8 +142,9 @@ def test_when_mock_time_sleep_then_interrupt_on_sleep(self, *args):
     @patch("cron.management.commands.run_cron.mock_cron_interrupt", side_effect=InterruptedError)
     def test_when_result_exists_in_given_interval_then_continue(self, *args):
         user = User.objects.create_user(username='test', email='test@test.com', password='test123')
```
[the remark this answers is not in the document]
Comment (andrew-susanto) [reply cron/tests.py:144]: okee otw delete, thank youu 🙏

File: error_logs/tests.py [hunk 3]
```diff
@@ -31,7 +32,10 @@ def test_when_non_authenticated_then_return_unauthorized(self):
   def test_when_authenticated_and_empty_data_then_empty_error_logs_success(self):
     # Create dummy user and authenticate
     user = User.objects.create_user(username='test', email='test@test.com', password='test123')
```
[the remark this answers is not in the document]
Comment (andrew-susanto) [reply error_logs/tests.py:34]: kalau disini masih kepake fii buat bikin tokennya wkwkk

File: login/views.py [hunk 4]
```diff
@@ -28,3 +30,45 @@ def user_login(request):
     
     return Response(data=data, status=status.HTTP_200_OK)
     
+
+@api_view(["POST"])
+@permission_classes([IsAuthenticated])
+def user_logout(request):
```
[the remark this answers is not in the document]
Comment (andrew-susanto) [reply login/views.py:36]: iya fii, ini sengaja dipindahin kesini buat rencana migrasi dari app logout nanti mau dijadiin satu app aja logout, login, dll

[the remark this answers is not in the document]
Comment (luthfifahlevi) [reply login/views.py:36]: oohh oke2, mantap

File: error_logs/tests.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (luthfifahlevi) [reply error_logs/tests.py:34]: oh iya deng wkwk halu pas baca. Sipp, thank youu

File: login/tests.py [hunk 5]
```diff
@@ -17,6 +20,42 @@ def test_when_non_authenticated_and_login_then_return_success(self):
...
+        self.assertEqual(tokens[0].team_member, team_member)
+        
+    def test_when_non_authenticated_and_team_exists_None_return_success(self):
+        # Create dummy user and authenticate
+        user = User.objects.create_user(username='test@gmail.com', email='test@gmail.com', password='Test1234')
+        team = Team.objects.create(name='test team')
+        team_member = TeamMember.objects.create(user=user, team=team)
+
+        request_params = {'email': 'test@gmail.com', 'password': 'Test1234'}
+        response = self.client.post(self.login_url, request_params)
```
[the remark this answers is not in the document]
Comment (andrew-susanto) [reply login/tests.py]: wkwkkw otw delete

**Review by luthfifahlevi — APPROVED** [review]
Very nice! Thank you

**Pull request merged**